### PR TITLE
Check if data-dir is a directory

### PR DESCRIPTION
**Pull request by mckennajones**
fix for #1838

### CODE REVIEW

**Comment by babbottscott** [comment]
imho, If an app is told to use a directory and has the perms to create it, it should attempt to create it if missing.  Nothing more frustrating than that one box out of thirty where someone made a boneheaded manual change that laid a timebomb your service can't recover from.

**Comment by mckennajones** [comment]
@babbottscott Thoughts on this @slackpad ? It wouldn't be hard to implement but I'm not sure if this is the intended behavior for this case.

**Comment by slackpad** [comment]
@mckennajones @babbottscott That's a good suggestion in general, but I'd keep the current behavior for Consul and not auto-create the directory. We use the presence of state in that directory to tell if a cluster is fresh and whether it's safe to bootstrap, for example, so I'd rather be conservative and have it error out. For non-servers it might be safe to create it on the fly, but that's probably too subtle/complex to have a behavior difference.

File: command/agent/command.go [hunk 1]
```diff
@@ -209,6 +209,12 @@ func (c *Command) readConfig() *Config {
 		return nil
 	}
 
+	if finfo, _ := os.Stat(config.DataDir); !finfo.IsDir() {
```
Comment (kyhavlov) [inline command/agent/command.go]: We should handle the error here in case they gave an invalid path.

Comment (mckennajones) [inline command/agent/command.go]: @kyhavlov What are you suggesting differently than what I have done?

**Pull request merged**